straxen/plugins/veto_events.py

```diff
@@ -0,0 +1,126 @@
+import strax
+import straxen
+
+import numpy as np
+import numba
+from immutabledict import immutabledict
+
+
+@strax.takes_config(
+    strax.Option('event_left_extension_nv', default=0,
+                 help="Extends events this many ns to the left"),
+    strax.Option('event_resolving_time_nv', default=300,
+                 help="Resolving time for fixed window concidene [ns]."),
+    strax.Option('event_min_hits_nv', default=6,
+                 help="Minimum number of fully convinet hitlets to define an event."),
+    strax.Option('gain_model_nv',
+                 help='PMT gain model. Specify as (model_type, model_config)'),
+    strax.Option('n_nveto_pmts', type=int,
+                 help='Number of TPC PMTs'),
+    strax.Option('channel_map', track=False, type=immutabledict,
+                 help="immutabledict mapping subdetector to (min, max) "
+                      "channel number."),
+)
+class nVETOEvents(strax.OverlapWindowPlugin):
+    """
+    Plugin which computes the boundaries of veto events.
+    """
+    depends_on = 'hitlets_nv'
+    provides = 'events_nv'
+
+    parallel = 'process'
+    compressor = 'zstd'
+    save_when = strax.SaveWhen.TARGET
+
+    __version__ = '0.0.1'
+    event_seen = 0
+
+    def setup(self):
+        self.to_pe = straxen.get_to_pe(self.run_id,
+                                       self.config['gain_model_nv'],
+                                       self.config['n_nveto_pmts'])
+
+        self.channel_range = self.config['channel_map']['nveto']
+
+    def infer_dtype(self):
+        return veto_event_dtype()
+
+    def get_window_size(self):
+        return self.config['event_left_extension_nv'] + self.config['event_resolving_time_nv'] + 1
+
+    def compute(self, hitlets_nv, start, end):
+        intervals = straxen.plugins.nveto_recorder.coincidence(hitlets_nv,
+                                                               self.config['event_min_hits_nv'],
+                                                               self.config['event_resolving_time_nv'],
+                                                               self.config['event_left_extension_nv']
+                                                               )
+
+        n_events = len(intervals)
+        events = np.zeros(n_events, dtype=veto_event_dtype())
```

Comment:
Is it worth passing `self.config['n_nveto_pmts']` here?

Reply:
Yes, good catch it is also needed if we want to inherit this method for the neutron veto.